### ISSUE TITLE
Add tooling to automate signing iOS/tvOS apps when running tests locally

### ISSUE THREAD

**Issue by steveisok** [dotnet/runtime#59717]
As of right now, the only way to run device tests locally is to first build the test so that the underlying xcode project is constructed.  Once that is done, you have to open Xcode, manually choose how you want to sign it, and then run.  This is rather cumbersome when you are trying to iterate over a test for whatever reason.  

To improve the experience, we should:

- [ ] Determine what properties need added / modified in the AppleAppBuilder task to support signing
- [ ] Add the automation hooks to support signing (look at Xam SDK and Xharness for inspiration)
- [ ] Document the steps

**Comment by premun** [comment]
@steveisok - just curious - what Apple account do you use for the signing? The large Microsoft one?

**Comment by mdh1418** [comment]
@premun I tried using the Microsoft team DevProsioningProfile to sign iOS apps to run on my iPad, but didn't get too far. I ended up using a personal profile and follow the steps
1) `./build.sh -s mono+libs -os iOS -arch arm64 -c Release`
2) ` ./dotnet.sh build /t:Test /p:TargetOS=iOS /p:TargetArchitecture=arm64 /p:Configuration=Release /p:DevTeamProvisioning=<my-personal-profile-provision> <path-to-test-project> (and this step errors out, but builds the .xcodeproj )
3) Open the `.xcodeproj` file, to which I've needed to go into signing & capabilities, chose my personal team, a unique bundle identifier and change signing certificate to `Sign to Run Locally`
4) Choose the scheme to run the app on my local device.

I'm going to look into and learn about the entire process and see whats needed. I hear that the solution might be to tweak the cmake files a bit.

**Comment by premun** [comment]
The process we use in Helix is something I came up with as there's not a lot of documentation how to automate this without a need for manual Xcode steps.
Let's meet and I can explain what's needed.

Meanwhile there's some documentation I wrote that captures some of this:
https://github.com/dotnet/core-eng/wiki/Apple-(iOS-tvOS)-signing-certificates

Here's the code that does the signing on the Helix machine:
https://github.com/dotnet/arcade/blob/6eb4672621de597024a7e287c82e870d03c0ab3f/src/Microsoft.DotNet.Helix/Sdk/tools/xharness-runner/xharness-runner.apple.sh#L71

Basically what you need is a keychain with the signing certificate and a provisioning profile file inside of the .app bundle. In the end it's quite simple but there's many gotcha's that are quite hard to debug/find.

**Comment by steveisok** [comment]
It appears `DevTeamProvisioning` works with a local build, which means once you set your team up in xcode, everything should "just work".  Closing.
